code/datums/elements/watery_tile.dm

```diff
@@ -0,0 +1,20 @@
+/datum/element/watery_tile
+	element_flags = ELEMENT_DETACH_ON_HOST_DESTROY
+
+/datum/element/watery_tile/Attach(turf/target)
+	. = ..()
+	if(!isturf(target))
+		return ELEMENT_INCOMPATIBLE
+
+	RegisterSignal(target, COMSIG_ATOM_ENTERED, PROC_REF(extinguish_atom))
+
+/datum/element/watery_tile/Detach(turf/source)
+	UnregisterSignal(source, COMSIG_ATOM_ENTERED)
+	return ..()
+
+/datum/element/watery_tile/proc/extinguish_atom(atom/source, atom/movable/entered)
+	SIGNAL_HANDLER
+
+	if(!isatom(entered))
+		return
```

Comment:
I am fairly certain nothing but atoms can `enter`, making this check redundant

Reply:
mfw atom_entered is a signal only sent by atoms